File: Notebooks/py/shakti17/fork-of-titanic/fork-of-titanic.py
#!/usr/bin/env python
# coding: utf-8

# In[ ]:


# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list the files in the input directory

import os
print(os.listdir("../input"))

# Any results you write to the current directory are saved as output.


# In[ ]:


train=pd.read_csv("../input/train.csv")
print(train.columns)


# In[ ]:


test=pd.read_csv("../input/test.csv")


# In[ ]:


from sklearn.preprocessing import Imputer
from sklearn.cross_validation import train_test_split
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.metrics import mean_absolute_error
data_train=train.copy()
var=["Pclass","Sex","Age"]
X=data_train[var]
Y=data_train.Survived
na_impute=Imputer()
X=pd.get_dummies(X)
X_impute=pd.DataFrame(na_impute.fit_transform(X))
X_impute.columns=X.columns
X_train,X_test,Y_train,Y_test=train_test_split(X_impute,Y,random_state=0)
clas=GradientBoostingClassifier()
clas.fit(X_train,Y_train)
print(mean_absolute_error(Y_test,clas.predict(X_test)))


# In[ ]:


X2=test[var]
X2=pd.get_dummies(X2)
X2_impute=pd.DataFrame(na_impute.fit_transform(X2))
X2_impute.columns=X2.columns
Y2=clas.predict(X2_impute)


# In[ ]:


Final=pd.DataFrame({"PassengerId": test['PassengerId'], "Survived": Y2})
Final.to_csv("Submission.csv",index=False)

File: Notebooks/r/liudj2008/fork-of-titanic/fork-of-titanic.R
# This R environment comes with all of CRAN preinstalled, as well as many other helpful packages

# The environment is defined by the kaggle/rstats docker image: https://github.com/kaggle/docker-rstats

# For example, here's several helpful packages to load in 



library(ggplot2) # Data visualization

library(readr) # CSV file I/O, e.g. the read_csv function



# Input data files are available in the "../input/" directory.

# For example, running this (by clicking run or pressing Shift+Enter) will list the files in the input directory



system("ls ../input")



# Any results you write to the current directory are saved as output.
#Load train data

train=read.csv('../input/train.csv')





train=train[,-c(1,4,9,11)]



#Fill the empty values

set.seed(1)

train$Age[is.na(train$Age)]=sample(train$Age[!is.na(train$Age)],177)



set.seed(2)

train$Embarked[train$Embarked=='']=sample(train$Embarked[train$Embarked!=''],2)

train$Embarked=as.character(train$Embarked)

train$Embarked=as.factor(train$Embarked)



train$Survived=as.factor(train$Survived)



summary(train)



# Modeling

library(randomForest)

set.seed(3)

rf.fit=randomForest(Survived~., data=train)

rf.pred=predict(rf.fit, newdata = train)



# Train error

table(true=train$Survived, pred=rf.pred)

mean(train$Survived==rf.pred)

# Load test sample

test_original=read.csv('../input/test.csv')



test=test_original



test=test[,-c(1,3,8,10)]

#summary(test)



# Initiate the survived factor and fill in the NAs

test=data.frame(Survived=as.factor(rep(0,418)), test)

set.seed(1)

test$Age[is.na(test$Age)]=sample(test$Age[!is.na(test$Age)],86)



set.seed(2)

test$Fare[is.na(test$Fare)]=sample(test$Fare[!is.na(test$Fare)],1)



# Making levels between training and testing data set is the same

names(test)

levels(test$Survived)=levels(train$Survived)

levels(test$Embarked)=levels(train$Embarked)



# Make prediction

test$Survived=predict(rf.fit, test)
# Export files

prediction=data.frame(PassengerId=test_original$PassengerId, Survived=test$Survived)





summary(prediction)

head(prediction)

# Output excel

write.csv(prediction,'gender_submission.csv')